### ISSUE TITLE
sql: cluster fails to reboot, "no allowed privileges found for system object" after restart with wrong binary

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#40702]
tldr: if I initialize a cluster with the latest `master` code, then try to restart with the `master` code from a few days ago, the cluster becomes hosed.

I think this is indicative of improper versioning somewhere. There should be guardrails against this mistake when made by users.

**Observations**

This is a freshly created cluster, created with `roachprod`:

```
$ roachprod create knz-aws-medium-h -u knz -c aws --aws-machine-type-ssd=c5d.4xlarge --local-ssd-no-ext4-barrier -n 7
# this installs binary from revision 99bed2718a9b5d1c6b94ce0d65dc40d1e746de16
$ roachprod stage knz-aws-medium-h cockroach 
$ roachprod start knz-aws-medium-h:1-6 --tag node
# (at this point the nodes are started OK)
$ roachprod stop knz-aws-medium-h:1-6 --tag node --sig 15 # notice the graceful stop
# staging a binary from revision 47bb2a58c87fc1259291ec9dde78de3e54bd8a3d
$ roachprod put knz-aws-medium-h:1-6 <my-binary> ./cockroach
$ roachprod start knz-aws-medium-h:1-6 --tag node
```

Here some the nodes fail, for example like this:

```
$ roachprod status knz-aws-medium-h:1-6
knz-aws-medium-h: status 6/6
   1: not running
   2: not running
   3: cockroach-v19.2.0-alpha.00000000-2955-gda20f35 3879
   4: cockroach-v19.2.0-alpha.00000000-2955-gda20f35 3920
   5: cockroach-v19.2.0-alpha.00000000-2955-gda20f35 3871
   6: cockroach-v19.2.0-alpha.00000000-2955-gda20f35 3818
```

Relevant log lines on node 1:
```
F190912 11:03:56.693471 57 server/server.go:1559  [n1] no allowed privileges found for system object with ID=25
failed to run migration "repeat: ensure admin role privileges in all descriptors"
github.com/cockroachdb/cockroach/pkg/sqlmigrations.(*Manager).EnsureMigrations
        /go/src/github.com/cockroachdb/cockroach/pkg/sqlmigrations/migrations.go:503
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start
        /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1553
github.com/cockroachdb/cockroach/pkg/cli.runStart.func3.2
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:698
github.com/cockroachdb/cockroach/pkg/cli.runStart.func3
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:813
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1337
```

Attached logs for the entire cluster. 
[init-fail.zip](https://github.com/cockroachdb/cockroach/files/3605184/init-fail.zip)

I also have the data directories if relevant.

After I shut down the remaining nodes, if I try to restart the cluster, other nodes fail to come up, with the same error.

**Comment by ajwerner** [comment]
I'm closing this as stale.
